src/main/java/mate/academy/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,175 @@
+                orders.add(getCopyOfOrder(resultSet));
+            }
+            LOGGER.info("Orders of user with id=" + user.getId() + " were received");
+            return orders;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of list of orders of user with id="
+                    + user.getId() + " is failed", e);
+        }
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO internet_shop.orders (user_id) values(?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+            addProductsToOrder(order);
+            LOGGER.info("Order with id=" + order.getId() + " was created");
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation of order is failed", e);
+        }
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * from internet_shop.orders where order_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCopyOfOrder(resultSet));
+            }
+            LOGGER.info("Order with id=" + id + " was received");
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of order with id="
+                    + id + " is failed", e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM internet_shop.orders;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(getCopyOfOrder(resultSet));
+            }
+            LOGGER.info("List of orders was received");
+            return orders;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of list of orders is failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM internet_shop.orders WHERE order_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
```

Comment:
if you are trying to delete order from orders but you have some rows with this orderId in orders_products what would happen then?

Reply:
It will remove all rows in other tables where is used order_id as foreign key. Thanks to   
ON DELETE CASCAD.